aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -116,6 +119,79 @@ def _get_multiple(self, path: str, **sdk_options) -> Dict[str, str]:
         """
         raise NotImplementedError()
 
+    def _set(
+        self,
+        name: str,
+        value: Union[str, dict, bytes],
+        *, # force keyword arguments
+        idempotency_id: Optional[str] = None,
+        version_stages: Optional[list[str]] = None,
+        create: bool = True,
+        **sdk_options
+    ) -> str:
+        """
+        Modifies the details of a secret, including metadata and the secret value.
+
+        Parameters
+        ----------
+        name: str
+            The ARN or name of the secret to add a new version to.
+        value: str or bytes
+            Specifies text data that you want to encrypt and store in this new version of the secret.
+        idempotency_token: str, optional
+            This value helps ensure idempotency. Recommended that you generate
+            a UUID-type value to ensure uniqueness within the specified secret.
+            This value becomes the VersionId of the new version. This field is
+            autopopulated if not provided.
+        version_stages: list[str], optional
+            Specifies a list of staging labels that are attached to this version of the secret.
+        create: bool, optional
+            Create the secret by default unless this is set to False, defaults to True
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the Secrets Manager update_secret API call
+
+        Returns:
+        -------
+            Version ID of the newly created version of the secret.
+
+        URLs:
+        -------
+            https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager/client/put_secret_value.html
+        """
+
+        sdk_options["SecretId"] = name
+
+        if isinstance(value, dict):
+            value = json.dumps(value)
+
+        if isinstance(value, bytes):
+            sdk_options["SecretBinary"] = value
+        else:
+            sdk_options["SecretString"] = value
+
+        if version_stages:
+            sdk_options["VersionStages"] = version_stages
+        if idempotency_id:
+            sdk_options["ClientRequestToken"] = idempotency_id
+
+        try:
+            value = self.client.put_secret_value(**sdk_options)
+            return value["VersionId"]
+        except ClientError as exc:
+            if exc.response["Error"]["Code"] != "ResourceNotFoundException":
+                raise SetParameterError(str(exc)) from exc
+            elif not create:
+                raise SetParameterError("Parameter does not exist, create before setting or set 'create' to True.")
+            else:
+                sdk_options.pop("SecretId")
+                sdk_options["Name"] = name
+                try:
+                    value = self.client.create_secret(**sdk_options)
+                    return value["VersionId"]
+                except Exception as exc:
+                    raise SetParameterError(str(exc)) from exc
```

Comment:
Can we simplify and comment on this code? I had to read 3 times to understand what it is doing. 

Also, I'm still not convinced if a unique method called `set_secret` is better than 2 methods called `update_secret` and `create_secret`. I don't know, but looking from the customers' side, I think the client can lose control of what they are creating/updating. This can incur extra costs and a bad user experience.

What do you think @heitorlessa @stephenbawks @aradyaron?

Reply:
I agree to the separation.
Secrets creation is, AFAiK, done in a different stage in the lifetime of the secret

Reply:
I do tend to lean towards the fact that we should set the `create` option to False by default.  

We want someone to specify when the secret should be created.  I think to @heitorlessa  point we would not want someone to have some process that is failing and being re-invoked and they happen to be using some random name for the secret and then they then have 193493 (exaggerating here a bit.... but possibly... ) secrets created for them.

I think they could certainly default `create` to be `False` and or catch the `SetParameterError` and THEN turn the `create` to `True`.

def handler(event, context):
    try:
        set_secret(....)
    except SetParameterErroras e:
        set_secret(...., create=True)
        ...

Reply:
For the time being I removed the create option.  Have to revisit that one.